argocd/resource_argocd_cluster.go

```diff
@@ -88,17 +88,16 @@ func resourceArgoCDClusterCreate(ctx context.Context, d *schema.ResourceData, me
 		}
 	}
 
-	client := *server.ClusterClient
+	tokenMutexClusters.Lock()
```

Comment:
Initially, I wanted to set an exclusive lock as late as possible, not to impact performance when a lot of clusters are being managed by ArgoCD. This will effectively slow down concurrent clusters creation. WDYT?

Reply:
:thinking:

Reply:
Been giving this some more thought. This definitely deserves a bit more discussion (and thought). So, for the time being, I figured I would break this PR down into smaller chunks (that don't necessarily require as much thought :smile:). I will provide a longer answer with my thoughts on this specific point ASAP.

Reply:
To be more specific, the lock release timing is crucial. I (vaguely) remember the test suite became instantly faster with the introduction of RLocks, so the assumption remained that TF blueprints handling large amounts of non thread-safe resources (tokens, and now clusters) may reach timeouts upon creation.

I realize this is an edge case, and don't have actual numbers to back that up. I'll leave that up to your discretion.